### DISCUSSION TITLE
Design v2

### DISCUSSION THREAD

**Discussion by chantastic** [LunchDevCommunity/community-calendar#93] [Ideas]
Hey friends!

I'd like to get the ball rolling on possible design changes.

I'm a 'form follows function' kinda designer so I'd like to approach it from the perspective of activities someone might like to perform on the site.

Right now I see 3 pretty major areas for improvement:

## Search and Categorization

That's technically two things but I think that their value is intertwined enough that doing them together makes sense.

Now that we have a backlog of events, it'd be great to categorize them in a way that they can be easily filtered.
For example, we have a good amount of eleventy content. Someone could reasonable build a site using our work as a tutorial.

Consequently, Episode Parties don't need to live on in perpetuity — unless they retain some form of long-term value. Now there actually might be some long-term value if we exported the highlights from our chat — similar to what I've done here (https://chan.dev/posts/single-threaded-episode-1-notes/).

But — without an artifact — past recurring events are mostly noise in the stream. It'd be nice to easily filter them out.

There are so many ways we could tackle this. Something like adding the YouTube video to front matter might even have the virtuous effect of making recorded events unfurl in social apps.

(stream of consciousness, but you get the ideas)

## Linking
There are resources that are not accessibly via anything but direct URL. These include:
* [Code of Conduct](https://events.lunch.dev/code-of-conduct/)
* [Sesh Instructions](https://events.lunch.dev/sesh/)
* [YouTube Channel](https://www.youtube.com/channel/UCB5V77aiP0pz7RVbfQrbr7Q)

I think that these links could start justifying a footer — though I'm not married to any outcome.

## Calendaring automation

This is the part I'm most unsure of implementation.

I'd like to make it much easier to add events. I'm not sure if that's a form that lives on the page — which triggers a GitHub Action and `sesh` event.
Or maybe it's the other way around and we catch `sesh` events and open PRs from there.
But I'd like to get these in some form of sync.

---

I'm excited to hear what other ideas you have as we start to make the content we're creating more accessible to others.

**Comment by BenDMyers** [comment]
I agree with all of this a lot. In my mind, there are two possible use cases for visiting the calendar:

1. To see what's coming up, in case calendars need to be marked.
2. To catch up on videos of old events.

Current use seems to be a whole lot of **1**, and much less **2**. I'd be curious if this is generally because of how people would _want_ to use a calendar like this (i.e. **1** is a much higher priority than **2** for most people), or because of a limitation of the current design. I suspect it's a bit of both, but understanding how we think people will actually use a tool like this will give us a good idea of what we need to prioritize for Design v2.

**Reply by chantastic** [reply]
this is a great way to consider the differences.

For `Case 1`, I've been thinking **Today, Next, This Week** distinction(s) could be a nice addition.

For `Case 2`, I think you're right. You're not so much browsing events. Wondering if we should lean into YouTube. Have a `Past events` section that is just the stream of YouTube videos. I'm sure we could get that via API.

A side-effect of this (which is ultimately a good thing) would be creating a habit of producing better thumbnails.

**Reply by chantastic** [reply]
I'm thinking thru this further and "ability to engage" is a big consideration.

Directing folks right to YouTube would allow them to use the comment section there — vs building comments into the eleventy site.

**Reply by BenDMyers** [reply]
I dig sending users directly to the YouTube uploads and skipping the intermediate past event page. Like we discuss in the other thread, the past event page should still exist, so we don't break links - the key difference is really that the homepage/event index pages would link externally.

**Comment by BenDMyers** [comment]
I totally agree we probably don't need to be cluttering the homepage with links to past episode parties — those are the least actionable, since - like you said - we don't create/upload artifacts like recordings to come back to.

I think I'd like to _hide_ past episode parties' links, but still leave the pages up — that way we aren't leaving a bunch of broken URLs in our wake.

**Reply by chantastic** [reply]
tooooootally. broken links are not a good practice.

this kinda ties into what i was thinking about with episode/stream events possible being in a JSON.
i don't know if it's important that these things be structurally different than events but it was a thought.

**Comment by fokusferit** [comment]
> I'd like to make it much easier to add events. I'm not sure if that's a form that lives on the page — which triggers a GitHub Action and sesh event.

I think that triggering a !create based on the data via a Github Action makes somehow more sense. I think it's easier to create a PR with some Markdown and merge it than understanding sesh bot correctly and creating the event 😄

**Reply by chantastic** [reply]
thank you! would you mind linking me to documentation @fokusferit?
